src/server/csp.ts

```diff
@@ -60,7 +67,7 @@ const assetHashes = (assets: string[]): string =>
 
 const buildCsp = ({ styles, scripts }: Assets, twitter: boolean): string => `
     default-src 'self';
-    style-src 'self' ${assetHashes(styles)} https://interactive.guim.co.uk ${twitter ? 'https://platform.twitter.com' : ''};
+    style-src 'self' 'unsafe-hashes' ${assetHashes(styles)} https://interactive.guim.co.uk ${twitter ? 'https://platform.twitter.com' : ''};
```

Comment:
Have you included this because there are `style` attributes on tags inside Chart atoms?

Reply:
yes, I managed to add the inline styles to the csp, but it looks like it still requires this.

https://github.com/guardian/apps-rendering/pull/701/files#diff-153287d02c1f9fa0719c0f23a8a8198eR150

Reply:
What does `unsafe-hashes` do when applied to `style-src`?

Reply:
There is not much information I could find, but it seems to enable inline style hashes. Without it, the inline style hashes are not accepted by the csp.

<img width="929" alt="Screen Shot 2020-09-18 at 14 22 58" src="">


| With | Without |
| --- | --- |
| <img src="https://user-images.githubusercontent.com/11618797/93602976-4c264b00-f9bb-11ea-9c4d-6fb760e06721.png" width="300" /> | <img src="https://user-images.githubusercontent.com/11618797/93603019-58aaa380-f9bb-11ea-86aa-2782a9d77094.png" width="300" /> |

Reply:
When `unsafe-hashes` *only* exists in `style-src`, not `script-src`, does it just apply to `style` attributes? Both event handlers and `javascript:` navigation targets remain forbidden?

Reply:
`unsafe-hashes` is CSP Level 3 which is only supported in Chrome and not in Safari so this won't work.

The only other thing I can think of is to match dotcom by doing
`script-src https: 'unsafe-inline';`